### ISSUE TITLE
Categories are lost when nesting in category manager

### ISSUE THREAD

**Issue by Charles-Garrison** [ExpressionEngine/ExpressionEngine#3832]
**Description of the problem**

**TLDR:** When moving a nesting a category under another category, all the nested category is removed from all assigned entries.

Let's say I have several categories in a group called "Digital Products", and that group is category ID 6.  Let's also say that I have an entry with a category set to "Backordered" as seen below.

In this case, my categories are structured like this:

- Paid Products
    - On Sale
    - Full Price
    - In Stock
- Backordered

If I then navigate to the category manager (/cp/categories/group/6), and I update my categories to the following, all entries in the "Backordered" category are now removed from that category.

- Paid Products
    - On Sale
    - Full Price
    - In Stock
    - Backordered  [now nested under "Paid Products"]

**How To Reproduce**
1. Create multiple categories
2. Assign a category to an entry and save
3. Go to the category manager and nest that category under one of the others
4. View the entry and see that the category was removed from the entry

**Error Messages**
None

**Screenshots / Videos / Template Code**
None

**Environment Details:**
- EE Version: 7.3.12
- Server OS: Ubuntu 22.04.3 LTS (jammy)
- PHP: 8.1.2-1ubuntu2.14, Zend Engine v4.1.2
- Mysql:  Ver 15.1 Distrib 10.6.12-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper
- Server version: Apache/2.4.52 (Ubuntu)

**Possible Solution**
Keep the relationship in-tact.

**Additional context**
None.

**Comment by intoeetive** [comment]
Hi @Charles-Garrison , thank you for the report. Would you be able to record video/screencast that would demonstrate the problem?

(FYI, not replicating this yet)

**Comment by Charles-Garrison** [comment]
Hi - I'm sorry, but now that I've tried to replicate several times myself, it's no longer happening for me either. If this happens again, I will investigate further and report again.
